reactive/kubernetes_master.py

```diff
@@ -1076,12 +1081,19 @@ def register_auth_webhook():
     if custom_authn:
         context["custom_authn_endpoint"] = custom_authn
 
+    k8s_log_path = Path(kubernetes_logs)
+    k8s_log_path.mkdir(parents=True, exist_ok=True)  # ensure log path exists
     render("cdk.master.auth-webhook-conf.yaml", auth_webhook_conf, context)
     render("cdk.master.auth-webhook.py", auth_webhook_exe, context)
     render(
         "cdk.master.auth-webhook.logrotate", "/etc/logrotate.d/auth-webhook", context
     )
 
+    # Move existing log files from ${auth_webhook_root} to /var/log/cdk/
+    for log_file in Path(auth_webhook_root).glob("auth-webhook.log*"):
+        if not (k8s_log_path / log_file.name).exists():
+            shutil.move(str(log_file), str(k8s_log_path))
```

Comment:
The new log file is `/var/log/kubernetes/cdk.master.auth-webhook.log`, but the move here places the old log file at `/var/log/kubernetes/auth-webhook.log`. Those are different - is this intentional?

Reply:
It's a very good question.  
I did this intentionally in the event that `filebeat` or other log collecting services would "see" the old file and collect its logs if it wasn't already collected.  I feared restarting the service while the file was in one location or the other.  this was an attempt to just "carry over" any logs or rotated during a charm upgrade without being destructive.

I'd value your opinion here though

Reply:
I guess my concern is that when someone comes along and looks in /var/log/kubernetes, on an upgraded cluster they might see:

```
$ ls -l /var/log/kubernetes
total 3784
-rw-r--r-- 1 root root   34138 Nov  5 20:17 auth-webhook.log
-rw-r--r-- 1 root root 3515014 Nov  8 16:37 cdk.master.auth-webhook.log
```

They might be confused about the difference between these two files, until they look at the dates and realize that auth-webhook.log is no longer being written. They have to *infer* that auth-webhook.log is an old file that's no longer being touched.

I would be tempted to add the `cdk.master.` prefix when moving the logs, to make the transition more seamless.

Reply:
A rename sounds in order.  

I'll test to make sure that the move|rename happens in such a way we do no overwrite a log currently in use by a an active service writing to it.

Reply:
Before Upgrade:
```bash
root@juju-958d60-0:~/cdk/auth-webhook# ls
__pycache__             auth-webhook.log       auth-webhook.log.10.gz  auth-webhook.log.3.gz  auth-webhook.log.5.gz  auth-webhook.log.7.gz  auth-webhook.log.9.gz
auth-webhook-conf.yaml  auth-webhook.log.1.gz  auth-webhook.log.2.gz   auth-webhook.log.4.gz  auth-webhook.log.6.gz  auth-webhook.log.8.gz  auth-webhook.py
```

After Upgrade:
```bash
root@juju-958d60-0:~/cdk/auth-webhook# ls
__pycache__  auth-webhook-conf.yaml  auth-webhook.py
root@juju-958d60-0:~/cdk/auth-webhook# ls /var/log/kubernetes/
cdk.master.auth-webhook.log       cdk.master.auth-webhook.log.10.gz  cdk.master.auth-webhook.log.3.gz  cdk.master.auth-webhook.log.5.gz  cdk.master.auth-webhook.log.7.gz  cdk.master.auth-webhook.log.9.gz
cdk.master.auth-webhook.log.1.gz  cdk.master.auth-webhook.log.2.gz   cdk.master.auth-webhook.log.4.gz  cdk.master.auth-webhook.log.6.gz  cdk.master.auth-webhook.log.8.gz
```